### PR TITLE
Migrate tests from Enzyme to React Testing Library

### PR DESCRIPTION
**Pull request by lambk**
# Description of changes

Converts tests from Enzyme to React Testing Library

Converted tests:

- CloseButton
- ErrorDisplay
- Modal
- OutboundLink
- ArticleGroup

# Issue Resolved

Works towards #850

### CODE REVIEW

**Comment by kylemh** [comment]
🤖 **This is a bot** 🤖<br /><br />🎉 Deployed Storybook preview! 🎉<br /><br />Click the link at the bottom of this comment to see it live.

Built with commit 5d5a1a1e05c3a16cc4eb6b984a03b4f84b3556c0

https://deploy-preview-863--operation-code-storybook.netlify.com

File: components/CloseButton/__tests__/CloseButton.test.js [hunk 1]
```diff
@@ -10,9 +10,9 @@ describe('CloseButton', () => {
 
   it('should not be clickable when disabled', () => {
     const onClickMock = jest.fn();
-    const wrapper = mount(<CloseButton disabled onClick={onClickMock} />);
+    const wrapper = render(<CloseButton disabled onClick={onClickMock} />);
 
-    wrapper.find('button').simulate('click');
+    fireEvent.click(wrapper.container.querySelector('button'));
```
Comment (kylemh) [inline components/CloseButton/__tests__/CloseButton.test.js]: Can you use `wrapper.queryByTestId`? I just added a `data-testid` to `Close Button`.

File: components/Press/PressLinks/ArticleGroup/__tests__/ArticleGroup.test.js [hunk 2]
```diff
@@ -28,7 +27,7 @@ describe('ArticleGroup', () => {
     ));
 
   it('should show all links when clicking Show All button', () => {
-    const ArticleGroupShallowInstance = mount(
+    const ArticleGroupShallowInstance = render(
```
Comment (kylemh) [inline components/Press/PressLinks/ArticleGroup/__tests__/ArticleGroup.test.js]: You could just destructure the `container` here

File: components/OutboundLink/__tests__/OutboundLink.test.js [hunk 3]
```diff
@@ -14,7 +13,7 @@ describe('OutboundLink', () => {
   });
 
   it('should render SVG when `hasIcon` is true', () => {
-    const wrapperWithIcon = mount(
+    const wrapperWithIcon = render(
```
Comment (kylemh) [inline components/OutboundLink/__tests__/OutboundLink.test.js]: can you change this to `component`?

File: components/OutboundLink/__tests__/OutboundLink.test.js [hunk 4]
```diff
@@ -25,11 +24,11 @@ describe('OutboundLink', () => {
       </OutboundLink>,
     );
 
-    expect(wrapperWithIcon.find('svg')).toExist();
+    expect(wrapperWithIcon.container.querySelector('svg')).not.toBeNull();
   });
 
   it('should not render SVG when `hasIcon` is false', () => {
-    const wrapperWithoutIcon = mount(
+    const wrapperWithoutIcon = render(
```
Comment (kylemh) [inline components/OutboundLink/__tests__/OutboundLink.test.js]: can you change this to `component`?

File: components/Press/PressLinks/ArticleGroup/__tests__/ArticleGroup.test.js [hunk 5]
```diff
@@ -40,27 +39,27 @@ describe('ArticleGroup', () => {
...
       />,
     );
 
-    expect(wrap.find('button').exists()).toStrictEqual(false);
+    expect(wrap.container.querySelector('button')).toBeNull();
   });
 
   it('should create a button if enough links are available', () => {
-    const wrap = mount(
+    const wrap = render(
```
Comment (kylemh) [inline components/Press/PressLinks/ArticleGroup/__tests__/ArticleGroup.test.js]: You could just `destructure` the container here

**Comment by lambk** [comment]
Sorry re delay, I will get to these after work 😃

File: components/OutboundLink/__tests__/OutboundLink.test.js (hunk 3 above)
Comment (lambk) [reply components/OutboundLink/__tests__/OutboundLink.test.js]: Sorry, what are you wanting changed here? (new to React testing library - couldn't find a component method)

Comment (kylemh) [reply components/OutboundLink/__tests__/OutboundLink.test.js]: Just the variable name. You can always just do:

`const component = render()`

or

`const { container } = render()` if you only need the container.

**Pull request merged**